akka-persistence-tck/src/main/scala/akka/persistence/snapshot/SnapshotStoreSpec.scala

```diff
@@ -106,5 +106,16 @@ abstract class SnapshotStoreSpec(config: Config) extends PluginSpec(config) {
       snapshotStore.tell(LoadSnapshot(pid, SnapshotSelectionCriteria(metadata(3).sequenceNr, metadata(3).timestamp), Long.MaxValue), senderProbe.ref)
       senderProbe.expectMsg(LoadSnapshotResult(Some(SelectedSnapshot(metadata(3), s"s-4")), Long.MaxValue))
     }
+    "save and overwrite snapshot with same sequence number" in {
+      val md = metadata(4)
+      snapshotStore.tell(SaveSnapshot(md, s"s-5-modified"), senderProbe.ref)
+      val md2 = senderProbe.expectMsgPF() { case SaveSnapshotSuccess(md2) ⇒ md2 }
+      md2.sequenceNr should be(md.sequenceNr)
+      snapshotStore.tell(LoadSnapshot(pid, SnapshotSelectionCriteria(md.sequenceNr), Long.MaxValue), senderProbe.ref)
+      val result = senderProbe.expectMsgType[LoadSnapshotResult]
+      result.snapshot.get.snapshot should be("s-5-modified")
+      result.snapshot.get.metadata.sequenceNr should be(md.sequenceNr)
+      // metadata timestamp may have been changed
+    }
```

Comment:
we don't overwrite in the file based store, but semantically a save followed by a load "looks like" an overwrite